p4-fusion/main.cc

```diff
@@ -63,9 +64,67 @@ int Main(int argc, char** argv)
 		return 1;
 	}
 
-	// Set the signal here because it gets reset after P4API library is initialized.
-	std::signal(SIGINT, SignalHandler);
-	std::signal(SIGTERM, SignalHandler);
+	// Block signals from being handled by the main thread, and all future threads.
+	//
+	// - SIGINT, SIGTERM, SIGHUP: thread pool will be shutdown and std::exit will be called
+	// - SIGUSR1: only sent by main() to tell the signal handler thread to exit
+	sigset_t blockedSignals;
+	sigemptyset(&blockedSignals);
+	for (auto sig : { SIGINT, SIGTERM, SIGHUP, SIGUSR1 })
+	{
+		sigaddset(&blockedSignals, sig);
+	}
+
+	int rc = pthread_sigmask(SIG_BLOCK, &blockedSignals, nullptr);
+	if (rc != 0)
+	{
+		ERR("(signal handler) failed to block signals: (" << errno << ") " << strerror(errno));
+		return 1;
+	}
+
+	// Create the thread pool
+	int networkThreads = arguments.GetNetworkThreads();
+	PRINT("Network Threads: " << networkThreads);
```

Comment:
can we leave this where it was? It feels nice to get a summary of all the settings in a single place.

Reply:
Do you mean the log line itself, or the declaration of the thread pool? I wanted to start the signal handling thread as close to the invocation of pthread_sigmask as possible, which is why I needed the threadpool construction to move up too.

Reply:
just this specific line 87 :)